code/modules/clothing/rogueclothes/mask/mask.dm

```diff
@@ -40,6 +40,20 @@
 		take_damage(11, BRUTE, "blunt", 1)
 	..()
 
+/obj/item/clothing/mask/rogue/spectacles/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if(!user.has_flaw(/datum/charflaw/badsight))
+		if(slot == SLOT_WEAR_MASK && user.STAPER < 5)
+			user.STAPER += 4
+			cured_eyesight = TRUE
+	user.update_fov_angles()
```

Comment:
```suggestion
	if(!user.has_flaw(/datum/charflaw/badsight) && slot == SLOT_WEAR_MASK && user.STAPER < 5)
		user.change_stat("perception", 4, "spectacles")
```
this is safer, and `change_stat("perception")` already includes the `update_fov_angles()` call. also this can be written with only one condition